app/assets/javascripts/category-picker.js

```diff
@@ -0,0 +1,189 @@
+;(function(root) {
+
+  "use strict";
+
+  var GOVUK = root.GOVUK;
+
+  function _unique(array) {
+    return $.grep(array, function(el, index) {
+        return index === $.inArray(el, array);
+    });
+  }
+
+  // wrapper around the categories data
+  var Categories = function (_categories) {
+    var primaries = []
+    var self = this
```

Comment:
I might be missing something here, but why are you assigning this to a self variable?